helix-term/src/ui/menu.rs

```diff
@@ -52,6 +54,40 @@ impl Item for PathBuf {
 
 pub type MenuCallback<T> = Box<dyn Fn(&mut Editor, Option<&T>, MenuEvent)>;
 
+pub struct FileChangeData {
+    pub cwd: PathBuf,
+    pub style_untracked: Style,
+    pub style_modified: Style,
+    pub style_deleted: Style,
+    pub style_renamed: Style,
+}
+
+impl Item for FileChange {
+    type Data = FileChangeData;
+
+    fn format(&self, data: &Self::Data) -> Row {
+        let process_path = |path: &PathBuf| {
+            path.strip_prefix(&data.cwd)
+                .unwrap_or(path)
+                .display()
+                .to_string()
+        };
+
+        let (sign, style, content) = match self {
+            Self::Untracked { path } => ("[+]", data.style_untracked, process_path(path)),
+            Self::Modified { path } => ("[~]", data.style_modified, process_path(path)),
+            Self::Deleted { path } => ("[-]", data.style_deleted, process_path(path)),
+            Self::Renamed { from_path, to_path } => (
+                "[>]",
+                data.style_modified,
+                format!("{} -> {}", process_path(from_path), process_path(to_path)),
+            ),
+        };
+
+        Row::new(vec![sign.to_owned(), content]).style(style)
```

Comment:
Styling the entire line seems like too much color to me. Could we color only the sign instead and leave the content as the default text color?

Reply:
For comparison, the current implementation:

![image](https://github.com/helix-editor/helix/assets/19556359/2ca5eb6a-67de-4df5-9083-4bff82e03aef)

Changing to only style the sign:

![image](https://github.com/helix-editor/helix/assets/19556359/193fd213-9e71-408d-8977-eb8407bdae23)

Personally I find the whole line highlight to be easier to work with. I can identify the types quicker. The original idea was to mimic the `git status` command where it's always the case that the whole line, including the change type letter(s) and path, is highlighted.

But as always definitely open to changes. Please let me know if I should make the change.

Reply:
I prefer whole line colors too

Reply:
I strongly prefer just the sign. @pascalkuthe @archseer thoughts?

Reply:
For the default theme I also prefer only coloring the sign but it may be ok in another theme.

We could just use two different theme keys here for the sign and the rest of the text so it's customizable (altough I would definitely only theme the sign in the default/"offical" themes).

Reply:
Okay lemme change to only style the signs unless @archseer thinks otherwise.